### DISCUSSION TITLE
Session Management with FastAPI_user

### DISCUSSION THREAD

**Discussion by Serjuvenal** [fastapi-users/fastapi-users#1385] [Q&A]
Hi!
I'm writing my first API, in Python, with FastAPI. I'm totally a newbie.

So,
This API needs User Authentication, after this happens, these authenticated users should be able to manipulate other entities in the DB through protected Endpoints and crud functions. Here is where I don't know how to use the same Assync Session from FastAPI_users.

**Comment by hgalytoby** [comment]
If you are following the full-example.

https://fastapi-users.github.io/fastapi-users/latest/configuration/full-example/#sqlalchemy

Then you can use the same Async Session with `get_async_session`.

https://github.com/fastapi-users/fastapi-users/blob/master/examples/sqlalchemy/app/db.py#L28

```py
@router.get('/foo')
async def foo(session: AsyncSession = Depends(get_async_session)):
    ...
```
